common/scala/src/main/scala/org/apache/openwhisk/core/database/elasticsearch/ElasticSearchActivationStore.scala

```diff
@@ -0,0 +1,482 @@
+
+    val res = client
+      .execute {
+        count(index) query { must(nameQuery, startRange) }
+      }
+      .map { res =>
+        if (res.status == StatusCodes.OK.intValue) {
+          val out = if (res.result.count > skip) res.result.count - skip else 0L
+          transid.finished(this, start, s"[COUNT] 'activations' completed: count $out")
+          JsObject(WhiskActivation.collectionName -> JsNumber(out))
+        } else {
+          transid.failed(this, start, s"Unexpected http response code: ${res.status}", ErrorLevel)
+          throw new Exception("Unexpected http response code: " + res.status)
+        }
+      }
+
+    reportFailure(
+      res,
+      failure =>
+        transid
+          .failed(this, start, s"[COUNT] 'activations' internal error, failure: '${failure.getMessage}'", ErrorLevel))
+  }
+
+  override def listActivationsMatchingName(
+    namespace: EntityPath,
+    name: EntityPath,
+    skip: Int,
+    limit: Int,
+    includeDocs: Boolean = false,
+    since: Option[Instant] = None,
+    upto: Option[Instant] = None,
+    context: UserContext)(implicit transid: TransactionId): Future[Either[List[JsObject], List[WhiskActivation]]] = {
+
+    val nameQuery = matchPhraseQuery("path", namespace.addPath(name).asString)
+    listActivations(namespace, skip, limit, nameQuery, includeDocs, since, upto, context)
+  }
+
+  override def listActivationsInNamespace(
+    namespace: EntityPath,
+    skip: Int,
+    limit: Int,
+    includeDocs: Boolean = false,
+    since: Option[Instant] = None,
+    upto: Option[Instant] = None,
+    context: UserContext)(implicit transid: TransactionId): Future[Either[List[JsObject], List[WhiskActivation]]] = {
+
+    val nameQuery = matchPhraseQuery("namespace", namespace.asString)
+    listActivations(namespace, skip, limit, nameQuery, includeDocs, since, upto, context)
+  }
+
+  private def listActivations(
+    namespace: EntityPath,
+    skip: Int,
+    limit: Int,
+    nameQuery: MatchPhrase,
+    includeDocs: Boolean = false,
+    since: Option[Instant] = None,
+    upto: Option[Instant] = None,
+    context: UserContext)(implicit transid: TransactionId): Future[Either[List[JsObject], List[WhiskActivation]]] = {
+
+    require(skip >= 0, "skip should be non negative")
+    require(limit >= 0, "limit should be non negative")
+
+    val start = transid.started(this, LoggingMarkers.DATABASE_QUERY, s"[QUERY] 'activations'")
+    val startRange = rangeQuery("start")
+      .gte(since.map(_.toEpochMilli).getOrElse(minStart))
+      .lte(upto.map(_.toEpochMilli).getOrElse(maxStart))
+    val index = generateIndex(namespace.namespace)
+
+    val res = client
+      .execute {
+        search(index) query { must(nameQuery, startRange) } sortByFieldDesc "start" limit limit from skip
+      }
+      .map { res =>
+        if (res.status == StatusCodes.OK.intValue) {
+          val out =
+            if (includeDocs)
+              Right(res.result.hits.hits.map { hit =>
```

Comment:
May be abstract the derserialize logic between `get` and `query` part when `includeDocs` is true